quickjs.c

```diff
@@ -48412,10 +48416,11 @@ void JS_AddIntrinsicMapSet(JSContext *ctx)
                                    js_map_proto_funcs_count[i]);
         obj1 = JS_NewCFunctionMagic(ctx, js_map_constructor, name, 0,
                                     JS_CFUNC_constructor_magic, i);
-        if (i < 2) {
-            JS_SetPropertyFunctionList(ctx, obj1, js_map_funcs,
-                                       countof(js_map_funcs));
-        }
+        if (i == 0)
+            JS_SetPropertyFunctionList(ctx, obj1, js_map_funcs, countof(js_map_funcs));
+        else if (i == 1)
+            JS_SetPropertyFunctionList(ctx, obj1, js_set_funcs, countof(js_set_funcs));
```

Comment:
Your code is not the least bit wrong but from a code clarity perspective, maybe consider assigning that `JS_CLASS_MAP + i` a few lines up to a local variable and then using it here:
```suggestion
        if (class_id == JS_CLASS_MAP)
            JS_SetPropertyFunctionList(ctx, obj1, js_map_funcs, countof(js_map_funcs));
        if (class_id == JS_CLASS_SET)
            JS_SetPropertyFunctionList(ctx, obj1, js_set_funcs, countof(js_set_funcs));
```

Reply:
👍👍